scheduler/group/group.go

```diff
@@ -0,0 +1,61 @@
+package group
+
+import (
+	"runtime"
+
+	"polydawn.net/repeatr/def"
+	"polydawn.net/repeatr/executor"
+	"polydawn.net/repeatr/lib/guid"
+	"polydawn.net/repeatr/scheduler"
+)
+
+// interface assertion
+var _ scheduler.Scheduler = &Scheduler{}
+
+// Dumb struct to send job references back
+type hold struct {
+	forumla  def.Formula
+	response chan def.Job
+}
+
+type Scheduler struct {
+	groupSize int
+	executor  *executor.Executor
+	queue     chan *hold
+}
+
+func (s *Scheduler) Configure(e *executor.Executor) {
+	s.groupSize = runtime.NumCPU()
+	s.executor = e
+	s.queue = make(chan *hold)
+}
+
+func (s *Scheduler) Start() {
+	for w := 1; w <= s.groupSize; w++ {
+		go s.Run()
+	}
+}
+
+func (s *Scheduler) Schedule(f def.Formula) <-chan def.Job {
+
+	h := &hold{
+		forumla:  f,
+		response: make(chan def.Job),
+	}
+
+	go func() {
+		s.queue <- h
+	}()
```

Comment:
Let's make this a queue.  It's currently not: this is a rube-goldberg use of goroutines that creates an implicit and unordered queue based on blocking otherwise pointless goroutines.

Making a buffered channel will actually perform the role of a queue.

We should also define behavior for when the queue is full.  This is an inevitable endgame for scheduler designs and does not get simpler later.  http://mechanical-sympathy.blogspot.co.uk/2012/05/apply-back-pressure-when-overloaded.html provides a good background on why; the "Rejected tasks" heading in https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ThreadPoolExecutor.html provides some very useful concrete examples of how other battle-hardened systems can deal with this.  For now a function for strategy and a default implementation that panics is probably fine, and is better than ignoring it.